### ISSUE TITLE
Namespaces: implicit port specification

### ISSUE THREAD

**Issue by dev-rke** [socketio/socket.io-client#877]
Hi,

i spent an hour to find out, that the port for the socket.io client is not respected, when i specify a namespace. My server runs on port 1337 and i wanted to connect with the following code:
```javascript
var socket = io('/my-namespace');
```
It did not work as expected.
I get notified in the browsers web developer console that cross origin requests are not allowed.
The request tries to fetch ```/socket-io/``` instead of ```:1337/socket-io/```.
My implementation follows the [offical guideline](http://socket.io/docs/rooms-and-namespaces/#custom-namespaces).

When i prepend a port to the code
```javascript
var socket = io(':1337/my-namespace');
```
everything works fine, connection establishes and i can transfer data.

So i request to respect a specified port implicitly when defining a namespace on clientside, because for protocol and servername it seems already to work.

**Comment by dev-rke** [comment]
I just found out, this might be a duplicate of #812.
But as mentioned here https://github.com/socketio/socket.io-client/commit/5abd97b2313aa3e9c9b2e1035dd4c24379fd77e8 the fix might not be available in V1.3.6.
I installed socket.io-client with npm as standalone, as dependency of socket.io and also via bower.
It seems that no released version contains the fix https://github.com/socketio/socket.io-client/commit/5abd97b2313aa3e9c9b2e1035dd4c24379fd77e8.

Could you please ensure that the fix will be included in the next release?

**Comment by nkzawa** [comment]
Fixed in 1.4.x
